### PR TITLE
feat: allow deep import of any package

### PR DESCRIPTION
**Pull request by Toxicable**
Allowos for correct npm label generation for deep imports of any package, not just @angular/material and @angular/cdk

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,6 +24,9 @@
     "@bazel/jasmine": "1.1.0",
     "@bazel/typescript": "1.1.0",
     "@types/jasmine": "3.5.0",
+    "@types/shelljs": "^0.8.6",
+    "@types/mock-fs": "^4.10.0",
+    "@bazel/bazel": "^2.0.0",
```
Comment (mattem) [inline package.json]: Can you remove the range selectors on the version number? We keep the deps as specific as possible.

Also the addition of the `@bazel/bazel` package seems out of scope here. We should pick a way to obtain bazel and go with it, currently that's bazelisk. If we think the bazel npm package is better suited we can switch.

File: src/generators/ts/ts.generator.ts [hunk 2]
```diff
@@ -106,20 +106,27 @@ export class TsGenerator extends BuildFileGenerator {
...
+    if (relative) {
       label = this.workspace.getLabelForFile(imp + '.ts');
       if (label) { return label; }
-    }
 
-    // fall back to assuming 3rd_party
-    return Label.parseAbsolute(`@${npmWorkspace}//${imp}`);
+      throw new Error (`Unable to generate label for: ${imp}`)
+    } else {
+      const pathParts = imp.split(sep);
```
Comment (mattem) [inline src/generators/ts/ts.generator.ts]: Won't the path always be separated with `/` regardless of the platform? `imp` holds the path in the module specifier.

**Review by mattem — CHANGES_REQUESTED** [review]

**Comment by mattem** [comment]
Thanks for the PR!
Regarding the `.vscode` dir, is that normally something that's committed? It's not something I'm overly familiar with.

File: package.json (hunk 1 above)
Comment (Toxicable) [reply package.json]: Ah right, we don't use bazelisk so not familar with it.
I think there's a issue in rules_nodejs to adjust @bazel/bazel to use it at some point though.
Anyway, ill remove it for now

**Comment by Toxicable** [comment]
@mattem Ah yeah, as usual, it depends.
We commit it at work since it has that funky config to fix up source maps, ill leave it out of this PR for now though since there's no isntructions on how to use it

File: src/generators/ts/ts.generator.ts (hunk 2 above)
Comment (Toxicable) [reply src/generators/ts/ts.generator.ts]: ah yes, this would have broken on windows otherwise

File: package.json [hunk 3]
```diff
@@ -24,6 +24,9 @@
     "@bazel/jasmine": "1.1.0",
     "@bazel/typescript": "1.1.0",
     "@types/jasmine": "3.5.0",
+    "@types/shelljs": "0.8.6",
+    "@types/mock-fs": "4.10.0",
+    "@bazel/bazel": "2.0.0",
```
Comment (mattem) [inline package.json]: Did we decide to remove `@bazel/bazel` for now?

File: src/generators/ts/ts.generator.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { parse } from 'path';
+import { parse, sep, join, posix } from 'path';
```
Comment (mattem) [inline src/generators/ts/ts.generator.ts]: I think `sep` is unused now that you switched to using `posix.sep`, so can be removed from the imports.

File: yarn.lock [hunk 5]
```diff
@@ -2,11 +2,42 @@
 # yarn lockfile v1
 
 
+"@bazel/bazel-darwin_x64@2.0.0":
```
Comment (mattem) [inline yarn.lock]: I think the lock file just needs to be regenerated after the last package.json changes

**Review by mattem — CHANGES_REQUESTED** [review]

**Comment by mattem** [comment]
I'm not sure why the GH actions aren't running, I'll dig into that, _don't think_ there is anything special that we have to do for forks.

File: package.json (hunk 3 above)
Comment (Toxicable) [reply package.json]: ah yes, all removed now

**Review by mattem — DISMISSED** [review]

**Comment by mattem** [comment]
Should have fixed the build action, can you rebase against master?

**Comment by Toxicable** [comment]
Ah awesome, will do this afternoon when I get home

**Comment by Toxicable** [comment]
@mattem fixed!

**Review by mattem — APPROVED** [review]

**Comment by mattem** [comment]
Looks good, thanks!
